pkg/cli/admin/waitfornoderestart/wait_for_node_restart.go

```diff
@@ -0,0 +1,281 @@
+package waitfornoderestart
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"k8s.io/klog/v2"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+
+	"github.com/openshift/oc/pkg/cli/admin/rebootmachineconfigpool"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/cli-runtime/pkg/resource"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+)
+
+var (
+	nodeKind = schema.GroupVersionKind{Group: "", Version: "v1", Kind: "Node"}
+)
+
+const (
+	RemoveOldTrustFieldManager = "remove-old-trust"
+)
+
+type WaitForNodeRestartRuntime struct {
+	ResourceFinder genericclioptions.ResourceFinder
+	KubeClient     kubernetes.Interface
+	DynamicClient  dynamic.Interface
+
+	RebootNumber int
+
+	masterRestartNumber int
+	workerRestartNumber int
+
+	genericclioptions.IOStreams
+}
+
+func (r *WaitForNodeRestartRuntime) setRestartNumber(ctx context.Context) error {
+	if r.RebootNumber > 0 {
+		r.masterRestartNumber = r.RebootNumber
+		r.workerRestartNumber = r.RebootNumber
+		return nil
+	}
+
+	masterMachineConfig, err := r.DynamicClient.Resource(rebootmachineconfigpool.MachineConfigResource).Get(ctx, rebootmachineconfigpool.MasterRebootingMachineConfigName, metav1.GetOptions{})
+	switch {
+	case apierrors.IsNotFound(err):
+	// do nothing.  If we need to use it, we'll fail
+	case err != nil:
+		return fmt.Errorf("unable to read existing state: %w", err)
+	default:
+		r.masterRestartNumber, err = rebootmachineconfigpool.GetRestartNumber(masterMachineConfig)
+		if err != nil {
+			return fmt.Errorf("unable to parse master restart number: %w", err)
+		}
+	}
+
+	workerMachineConfig, err := r.DynamicClient.Resource(rebootmachineconfigpool.MachineConfigResource).Get(ctx, rebootmachineconfigpool.WorkerRebootingMachineConfigName, metav1.GetOptions{})
+	switch {
+	case apierrors.IsNotFound(err):
+	// do nothing.  If we need to use it, we'll fail
+	case err != nil:
+		return fmt.Errorf("unable to read existing state: %w", err)
+	default:
+		r.workerRestartNumber, err = rebootmachineconfigpool.GetRestartNumber(workerMachineConfig)
+		if err != nil {
+			return fmt.Errorf("unable to parse worker restart number: %w", err)
+		}
+	}
+
+	return nil
+}
+
+func (r *WaitForNodeRestartRuntime) Run(ctx context.Context) error {
+	if err := r.setRestartNumber(ctx); err != nil {
+		return err
+	}
+
+	hasMaster := false
+	hasWorker := false
+	remainingNodes := []*corev1.Node{}
```

Comment:
remainingNodes end up being all nodes right? Or did you want this to be a subset of nodes?